src/requestVariableCompletionItemProvider.ts

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+import { CompletionItemProvider, TextDocument, Position, CancellationToken, CompletionItem, CompletionItemKind, Range, TextLine } from 'vscode';
+
+import { RequestVariableCacheValueProcessor } from "./requestVariableCacheValueProcessor";
+import { VariableProcessor } from "./variableProcessor";
+import { VariableUtility } from "./variableUtility";
+import * as Constants from "./constants";
+
+const firstPartRegex: RegExp = /^(\w+)\.$/;
+const secondPartRegex: RegExp = /^(\w+)\.(request|response)\.$/;
+
+export class RequestVariableCompletionItemProvider implements CompletionItemProvider {
+    public async provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken): Promise<CompletionItem[]> {
+        if (!VariableUtility.isPartialRequestVariableReference(document, position)) {
+            return [];
+        }
+
+
+        const wordRange = document.getWordRangeAtPosition(position, /\{\{(\w+)\.(.*?)?\}\}/);
+        let lineRange = document.lineAt(position);
+
+        let fullPath = this.getRequestVariableCompletionPath(wordRange, lineRange, position);
+
+        if (!fullPath) {
+            return undefined;
+        }
+
+        const match = fullPath.match(/(\w+)\.(.*?)?/);
+        if (!match || !this.checkIfRequestVariableDefined(document, match[1])) {
+            return [];
+        }
+
+        if (firstPartRegex.test(fullPath)) {
+            return [
+                new CompletionItem("request", CompletionItemKind.Field),
+                new CompletionItem("response", CompletionItemKind.Field),
+            ];
+        } else if (secondPartRegex.test(fullPath)) {
+            return [
+                new CompletionItem("body", CompletionItemKind.Field),
+                new CompletionItem("headers", CompletionItemKind.Field),
+            ];
+        }
+
+        let completionItems: CompletionItem[] = [];
+
+        const fileRequestVariables = VariableProcessor.getRequestVariablesInFile(document);
+        for (let [variableName, variableValue] of fileRequestVariables) {
+            let regex = new RegExp(`(${variableName})\.(request|response)\.(body|headers)\..*`);
+            if (regex.test(fullPath)) {
+                // Remove last dot if present
+                fullPath =
+                    fullPath.charAt(fullPath.length - 1) === "."
+                        ? fullPath.substring(0, fullPath.length - 1)
+                        : fullPath;
+
+                const valueAtPath = RequestVariableCacheValueProcessor.getValueAtPath(variableValue, fullPath);
+                if (valueAtPath) {
+                    let props = Object.getOwnPropertyNames(valueAtPath);
+
+                    completionItems = props.map(p => {
+                        let item = new CompletionItem(p);
+                        item.detail = `(property) ${p}`;
+                        item.documentation = p;

```

Comment:
How about make the value of `documentation` to the actual value of `p`? User can better know what's the actual value will be inserted into the request.